src/common/snippets/src/pass/propagate_precision.cpp

```diff
@@ -0,0 +1,289 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/pass/propagate_precision.hpp"
+
+#include <assert.h>
+#include <memory>
+#include "ov_ops/type_relaxed.hpp"
+#include "snippets/itt.hpp"
+#include "ngraph/rt_info.hpp"
+
+using namespace ngraph;
+
+ngraph::snippets::pass::PropagatePrecision::PropagatePrecision(
+    const ov::element::Type supported_precision,
+    const std::shared_ptr<const TargetMachine>& target_machine) : supported_precision(supported_precision), target_machine(target_machine) {
+}
+
+bool ngraph::snippets::pass::PropagatePrecision::run_on_model(const std::shared_ptr<ov::Model>& f) {
+    RUN_ON_MODEL_SCOPE(PropagatePrecision);
+    OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::op::PropagatePrecision")
+
+    std::unordered_map<std::shared_ptr<ngraph::opset1::Result>, element::Type> result_types;
+    auto results = f->get_results();
+    for (auto& result : results) {
+        result_types.emplace(result, result->get_input_source_output(0).get_element_type());
+    }
+
+    for (const auto& op : f->get_ordered_ops()) {
+        if (ngraph::is_type<opset1::Constant>(op)) {
```

Comment:
What's about `Result`, `Parameter`?

Reply:
We don't need to check operation type here. `Constant` check was removed too.